source/common/router/upstream_request.cc

```diff
@@ -92,7 +92,8 @@ UpstreamRequest::~UpstreamRequest() {
         std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time_);
     parent_.cluster()
         ->timeoutBudgetStats()
```

Comment:
here don't see quarding if `timeoutBudgetStats() != nullopt`

Reply:
It is handled using `record_timeout_budget_` in [Line 88](https://github.com/envoyproxy/envoy/blob/e6e61ae66773b603452f13230fd742f7e0ef5251/source/common/router/upstream_request.cc#L88) which is only set if  `timeoutBudgetStats() != nullopt` [here](https://github.com/envoyproxy/envoy/blob/e6e61ae66773b603452f13230fd742f7e0ef5251/source/common/router/upstream_request.cc#L53).